portfolio/src/main/webapp/script.js

```diff
@@ -40,4 +40,33 @@ function KaylaTrivia(){
     else{
          alert(`Your final score is ${points}  Oof maybe try again?`);
     }
+}
+function getMessage() {
+  console.log('Fetching a random quote.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addQuoteToDom);
+}
+
+/** Adds a random quote to the DOM. */
```

Comment:
Nice job leaving code comments! Small recommendation: Try adding to your jsdoc to include the `@param` annotation: https://google.github.io/styleguide/jsguide.html#jsdoc-general-form

Reply:
Also feel free to update the description. It doesn't look like it adds a random quote to the DOM -- it adds the quote that the function was given as an argument to the DOM :)